packages/squiggle-lang/__tests__/TS/Scalars_test.ts

```diff
@@ -0,0 +1,51 @@
+import { errorValueToString } from "../../src/js/index";
+import { testRun } from "./TestHelpers";
+import * as fc from "fast-check";
+
+describe("Scalar manipulation is well-modeled by javascript math", () => {
+  test("in the case of logarithms (with assignment)", () => {
+    fc.assert(
+      fc.property(fc.float(), (x) => {
+        let squiggleString = `x = log(${x}); x`;
+        let squiggleResult = testRun(squiggleString);
+        if (x == 0) {
+          expect(squiggleResult.value).toEqual({
+            tag: "number",
+            value: -Infinity,
+          });
+        } else if (x < 0) {
+          expect(squiggleResult.value).toEqual({
+            tag: "RETodo",
+            value:
+              "somemessage (confused why a test case hasn't pointed out to me that this message is bogus)",
+          });
+        } else {
+          expect(squiggleResult.value).toEqual({
+            tag: "number",
+            value: Math.log(x),
+          });
+        }
+      })
+    );
+  });
+
+  test("in the case of addition (with assignment)", () => {
+    fc.assert(
+      fc.property(fc.float(), fc.float(), fc.float(), (x, y, z) => {
+        let squiggleString = `x = ${x}; y = ${y}; z = ${z}; x + y + z`;
+        let squiggleResult = testRun(squiggleString);
+        switch (squiggleResult.tag) {
+          case "Error":
+            expect(errorValueToString(squiggleResult.value)).toEqual(
+              "some message (hopefully a test case points it out to me)"
```

Comment:
Shouldn't this not fail?

Reply:
This is yet another instance of checking a switch case that I didn't want to check because typescript wouldn't let me take `squiggleResult.value.value` (until of course I relented and put an `any` in `testRun` return type--- inconsistent style across this PR is due to making that decision late in the game!)

Reply:
You could also use an ``toExn`` function or similar. (And if there's not one yet, it could be good to make one, even with ``any`` if needed.)

Reply:
irrelevant after revamp in next push.